security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -552,6 +532,52 @@ func (sc *SecretController) scrtUpdated(oldObj, newObj interface{}) {
 	}
 }
 
+// tryToSyncKeyCertBundle tries to sync root cert in keycertbundle with root
+// cert from istio-ca-secret. Returns error if any step fails.
+func (sc *SecretController) tryToSyncKeyCertBundle(scrtName string, rootCertInMem, caCertInMem,
+	rootCertInWorkloadScrt []byte) ([]byte, error) {
+	// Check if root certificate in key cert bundle is not up-to-date. With multiple
+	// Citadel deployed in Istio, the root certificate in istio-ca-secret could be
+	// rotated by any Citadel and become newer than the one in local key cert bundle.
+	// Add a 30 seconds interval for key cert bundle sync to prevent I/O burst.
+	if sc.lastKCBSyncTime.IsZero() || time.Since(sc.lastKCBSyncTime) > 30*time.Second {
```

Comment:
Make the false condition check upfront?
```
if !sc.lastKCBSyncTime.IsZero() && time.Since(sc.lastKCBSyncTime) < 30*time.Second {
    rootCertInMem, nil
}
```